dorado/routines.py

```diff
@@ -794,3 +815,129 @@ def plot_state(grid, walk_data, iteration=-1, target_time=None, c='b'):
     plt.scatter(y, x, c=c)
 
     return ax
+
+
+def snake_plots(particle,
+                num_steps,
+                folder_name=None,
+                interval=4,
+                tail_length=12,
+                rgba_start=[1, 0.4, 0.2, 1],
+                rgba_end=[1, 0.3, 0.1, 0]):
+    """Plot particle positions with a trailing tail
+    
+    Loops through existing walk_data history and creates a series of
+    plots of the particle locations, with recent locations shown as a
+    trailing tail. Default is for the tails to fade away, but they 
+    can also change color.
+
+    **Inputs** :
+
+        particle : :obj:`dorado.particle_track.Particles`
+            An initialized :obj:`particle_track.Particles` object with an
+            existing walk history stored in particle.walk_data, as well as
+            a particle.depth array to serve as the background
```

Comment:
I think it be preferable to have this take `all_walk_data` as an input instead. Then we could save the walk data from a run, load it later and still use this function (similar to how we set up the example for the `draw_travel_path()` function).

Then, similar to the change made to `draw_travel_path()` in this PR, a new input for the background grid would have to be specified. The upside is that this would allow the user to specify the background data they wish to plot the particles on top of (maybe topography or velocity instead of depth). Ideally this input should be flexible and allow for a 3-D (time-x-y) input to cover unsteady simulations where the depth and/or velocity fields have varied over the course of the particle routing. Not sure how simple or tricky that would be to implement, but this function is a bit more flexible than the `steady_plots()` routine which ensures that the particle simulation is "steady" as it handles the transport internally. 

If that seems like it isn't easy or feasible to implement, I think it would be fine to instead add a note to the documentation to make it clear that the plots created by this routine use a fixed 2D array as a background image.

Reply:
You're right, this is a good call. Will detach the inputs from the particle object.

The issue I'm having with making the background time-varying is that syncing the plots up with `travel_time` rather than iterations is proving to be a bit of a nightmare. It's currently built to plot by iterations because slicing the previous locations (e.g. find the location 5 iterations ago) is trivial in comparison to time (e.g. find where this particle was at 5 min ago, and 10 min ago, and hopefully those aren't the same location or it'll look funny, etc). 

My plan is to either modify `snake_plots` or create an equivalent `snake_time_plots` down the line once I can figure out a simple way to implement it. But I want to do a bit more testing on the nuances of the `travel_time` computation itself before working on that anyway. Once that's done, it should be able to handle unsteady cases.

Reply:
That sounds good. I think the updated function is well described (handles iterations, the `grid` object is the 2-D background).